### DISCUSSION TITLE
Guidance on upgrading from elmah and elmah.corelibrary to ElmahCore

### DISCUSSION THREAD

**Discussion by behm** [ElmahCore/ElmahCore#198] [General]
We are using the following NuGet libraries and are looking for guidance on how to upgrade to ElmahCore.  Our application is currently written in .NET Framework 4.7.1.

- https://www.nuget.org/packages/elmah
- https://www.nuget.org/packages/elmah.corelibrary

We mostly make use of the ErrorSignal and ErrorLog classes in the custom logger method below and are looking for examples of equivalent usage in ElmahCore:

```
public virtual void LogException(Exception ex)
{
    try
    {
        if (HttpContext.Current != null)
        {
            ErrorSignal.FromCurrentContext().Raise(ex);
        }
        else
        {
            ErrorLog.GetDefault(null).Log(new Error(ex));
        }

        TrackExceptionWithAppInsights(ex);
    }
    catch (Exception)
    {
    }
}
```

Our application is currently written in .NET Framework 4.7.1.

Any guidance you are able to provide would be much appreciated.

**Comment by fraserriver** [comment]
@behm I am also looking for guidance here. Specifically, what is the replacement for ErrorLog.GetDefault() since ElmahCore does not have this method. Did you figure this out yet?
